Reusing a shared Artemis container in Quarkus Devservice not working properly

When a shared container should be used the URI is missing any scheme.

As io.quarkus.devservices.common.ContainerAddress.getUrl() method returns only a String containing "host:port".

https://github.com/quarkiverse/quarkus-artemis/blob/b3800a4f1bc68764dd1c805547d5d8d4e3908a4d/core/deployment/src/main/java/io/quarkus/artemis/core/deployment/DevServicesArtemisProcessor.java#L259-L265

Subsequently it fails, as the returned URI is not valid.

```
2024-10-12 13:05:58,639 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:119)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:116)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.IllegalStateException: java.lang.RuntimeException: java.lang.NullPointerException: Schema localhost not found
        at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.<init>(ActiveMQConnectionFactory.java:239)
        at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.<init>(ActiveMQConnectionFactory.java:280)
        at io.quarkus.artemis.jms.runtime.ArtemisJmsRecorder.extractConnectionFactory(ArtemisJmsRecorder.java:52)
        at io.quarkus.artemis.jms.runtime.ArtemisJmsRecorder.getConnectionFactoryProducer(ArtemisJmsRecorder.java:35)
        at io.quarkus.deployment.steps.ArtemisJmsProcessor$configure1243796012.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.ArtemisJmsProcessor$configure1243796012.deploy(Unknown Source)
        ... 11 more
Caused by: java.lang.RuntimeException: java.lang.NullPointerException: Schema localhost not found
        at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.setBrokerURL(ActiveMQConnectionFactory.java:260)
        at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.<init>(ActiveMQConnectionFactory.java:237)
        ... 16 more
Caused by: java.lang.RuntimeException: java.lang.NullPointerException: Schema localhost not found
        at org.apache.activemq.artemis.core.client.impl.ServerLocatorImpl.newLocator(ServerLocatorImpl.java:362)
        at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.setBrokerURL(ActiveMQConnectionFactory.java:257)
        ... 17 more
Caused by: java.lang.NullPointerException: Schema localhost not found
        at org.apache.activemq.artemis.utils.uri.URIFactory.newObject(URIFactory.java:56)
        at org.apache.activemq.artemis.core.client.impl.ServerLocatorImpl.newLocator(ServerLocatorImpl.java:360)
        ... 18 more
```

--- a/core/deployment/src/main/java/io/quarkus/artemis/core/deployment/DevServicesArtemisProcessor.java
+++ b/core/deployment/src/main/java/io/quarkus/artemis/core/deployment/DevServicesArtemisProcessor.java
@@ -262,7 +262,7 @@ public class DevServicesArtemisProcessor {
                         containerAddress.getId(),
                         null,
                         urlPropertyName,
-                        containerAddress.getUrl()))
+                        String.format("tcp://%s", containerAddress.getUrl())))
                 .orElseGet(defaultArtemisBrokerSupplier);
     }
 

